scio-elasticsearch/src/main/java/com/spotify/scio/elasticsearch/ElasticsearchIO.java

```diff
@@ -0,0 +1,228 @@
+     * slight delay specified by flushInterval.
+     *
+     * @param flushInterval delay applied to buffer elements. Defaulted to 1 seconds.
+     */
+    public static Bound withFlushInterval(Duration flushInterval) {
+      return new Bound().withFlushInterval(flushInterval);
+    }
+
+    public static class Bound extends PTransform<PCollection<IndexRequestWrapper>, PDone> {
+      private final String clusterName;
+      private final InetSocketAddress[] servers;
+      private final Duration flushInterval;
+
+      private Bound(final String clusterName,
+                    final InetSocketAddress[] servers,
+                    final Duration flushInterval) {
+        this.clusterName = clusterName;
+        this.servers = servers;
+        this.flushInterval = flushInterval == null? Duration.ofSeconds(1L): flushInterval;
+      }
+
+      Bound() {
+        this(null, null, null);
+      }
+
+      public Bound withClusterName(String clusterName) {
+        return new Bound(clusterName, servers, flushInterval);
+      }
+
+      public Bound withServers(InetSocketAddress[] servers) {
+        return new Bound(clusterName, servers, flushInterval);
+      }
+
+      public Bound withFlushInterval(Duration flushInterval) {
+        return new Bound(clusterName, servers, flushInterval);
+      }
+
+      @Override
+      public PDone expand(final PCollection<IndexRequestWrapper> input) {
+
+        if (clusterName == null) {
+          throw new IllegalStateException(
+              "need to set clustername of ElasticsearchIO.Write transform");
+        }
+
+        if (servers == null) {
+          throw new IllegalStateException(
+              "need to set clustername of ElasticsearchIO.Write transform");
+        }
+
+        input
+            .apply("Assign To Shard", ParDo.of(new AssignToShard()))
+            .apply("Re-Window to Global Window", Window.<KV<Long, IndexRequestWrapper>>into(new GlobalWindows())
+                       .triggering(Repeatedly.forever(
+                           AfterProcessingTime
+                               .pastFirstElementInPane()
+                               .plusDelayOf(javaToJoda(flushInterval))))
+                       .discardingFiredPanes())
+            .apply(GroupByKey.create())
+            .apply("Write to Elasticesarch", ParDo.of(new ElasticsearchWriter(clusterName, servers)));
+        return PDone.in(input.getPipeline());
+      }
+      private org.joda.time.Duration javaToJoda(final Duration duration) {
+        return duration == null ? null : org.joda.time.Duration.millis(duration.toMillis());
+      }
+    }
+
+    private static class AssignToShard
+        extends DoFn<IndexRequestWrapper, KV<Long, IndexRequestWrapper>> {
+      private int numWorkers;
+
+      @StartBundle
+      public void startBundle(Context c) throws Exception {
+        numWorkers = c
+            .getPipelineOptions()
+            .as(DataflowPipelineOptions.class)
```

Comment:
@ravwojdyla Would going higher in interface make it close to agnostic? The function `getNumWorkers` is defined in `DataflowPipelineWorkerPoolOptions`. All the option class should be inheriting from this class so it might generalize it? Wouldn't all runner require same/similar options?